### PR TITLE
Clarify implicit and explicit dependencies

### PR DESCRIPTION
**Pull request by plaindocs**
Improve docs in response to #714

### CODE REVIEW

**Comment by harrietgrace** [comment]
Great addition 🙌🏻Do you think we should add notes about this elsewhere too? Like on the dependencies page?

**Comment by plaindocs** [comment]
Hey @harrietgrace 

You mean more than the note that I added already? :-p

I did wonder about adding a section about the interactions, and looping in `depends_on: ~` at @matthewd 's sugggestion.

I was also confused by 

> There are three step attributes that can each affect when a step is able to run: if/branches, depends_on, and concurrency_group. They are applied to the step in the order they are listed.

So will tweak that

File: pages/pipelines/dependencies.md.erb [hunk 1]
```diff
@@ -90,12 +90,18 @@ Even though the second command step in the above example is after a wait step, t
...
 ## Order of operations
 
-There are three step attributes that can each affect when a step is able to run: `if`/`branches`, `depends_on`, and `concurrency_group`. They are applied to the step in the order they are listed.
+There are three step attributes that can each affect when a step is able to run:
+
+* `if`/`branches`
+* `depends_on`
+* `concurrency_group`
 
 If the step you're dependent on doesn't exist, the build will fail without running the step that is waiting for the dependency.
```
Comment (plaindocs) [inline pages/pipelines/dependencies.md.erb:99]: @harrietgrace here do you mean if it excluded by a branches condition?

File: pages/pipelines/dependencies.md.erb [hunk 2]
```diff
@@ -90,12 +90,18 @@ Even though the second command step in the above example is after a wait step, t
...
 
 However, if the step you're dependent on is excluded from the build due to an `if` condition, the dependency will be ignored and the step that depends on it will run once any other dependencies are satisfied.
 
+Steps that are in a `concurrency_group` run in the order they are created in, and so if a step depends on another step in a concurrency `concurrency_group` that will affect when the first step is able to run.
```
Comment (plaindocs) [inline pages/pipelines/dependencies.md.erb:103]: Is this what you had in mind for the third bullet point?

File: pages/pipelines/dependencies.md.erb (hunk 1 above)
Comment (harrietgrace) [reply pages/pipelines/dependencies.md.erb:99]: Yep, or if you mis-typed the name, or a pipeline upload didn't work... anything that might cause a step to not be there.

File: pages/pipelines/dependencies.md.erb (hunk 2 above)
Comment (harrietgrace) [reply pages/pipelines/dependencies.md.erb:103]: Mmm, not quite? Concurrency groups are the worst for explaining 😅

How about something that goes a little deeper, along the lines of:

```suggestion
Steps that are in a `concurrency_group` run in the order they are created in and can be delayed in running by the `concurrency` attribute. If your step has a dependency on a step that is in a `concurrency_group`, there is an implicit dependency on the rest of the steps in the group. For more information about concurrency groups, see the [Controlling Concurrency guide](https://buildkite.com/docs/pipelines/controlling-concurrency#concurrency-groups).
```

Comment (plaindocs) [reply pages/pipelines/dependencies.md.erb]: Perfect - thanks

**Pull request merged**